### PR TITLE
Dataiku integration tip

### LINKED ISSUE #830: Add Dataiku integration tip
**Description**
Add an article like the one for Databricks but for Dataiku.

**Expected change**
Create a new tip/article for Dataiku

### PR DESCRIPTION
**Pull request by FlorianJacta**
Fixes: #830 

Unzip this file to see the built tip:

[Uploading Integrating with Dataiku - Taipy (2).zip…]()

### CODE REVIEW

File: docs/knowledge_base/tips/dataiku/index.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
+---
+title: Integrating Dataiku
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
title: Integrating with Dataiku
```

File: docs/knowledge_base/tips/dataiku/index.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+category: tips
+type: code
+data-keywords: scenario task
+short-description: A guide to integrate Dataiku with Taipy.
+img: dataiku/images/dataiku.png
+---
+
+Integration with external platforms is often essential for executing tasks and functions efficiently. This article presents a comprehensive guide on integrating Dataiku with Taipy, a tool for orchestrating tasks and exploring various iterations of business problems. By integrating Dataiku projects with Taipy, users can enhance their data processing and visualization capabilities.
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
Integration with external platforms is often essential for executing tasks and functions
efficiently. This article presents a comprehensive guide on integrating Dataiku with Taipy,
a tool for orchestrating tasks and exploring various iterations of business problems. By
integrating Dataiku projects with Taipy, users can enhance their data processing and
visualization capabilities.
```

File: docs/knowledge_base/tips/dataiku/index.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+![Dataiku](images/dataiku.png){width=90% : .tp-image}
+
+# Simplified Integration: Streamlining Metric Visualization
+
+## Setting Up Your Environment
+
+Begin by ensuring that you have the required libraries installed. You'll need `dataiku-api-client` for accessing Dataiku and `taipy-gui` for building the GUI. Install these libraries using pip:
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
Start by ensuring that you have the required libraries installed. You'll need 
`dataiku-api-client` to access Dataiku and `taipy-gui` to build the GUI. Install these 
libraries using pip:
```

File: docs/knowledge_base/tips/dataiku/index.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+```bash
+pip install dataiku-api-client taipy pandas
+```
+
+## Connecting to Dataiku
+
+Use the `DSSClient` from the `dataikuapi` package to establish a connection with your Dataiku instance. Replace `"http://your-dataiku-instance.com"` and `"your_api_key"` with your actual Dataiku host and API key.
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
Use the `DSSClient` from the `dataikuapi` package to establish a connection with your Dataiku
instance. Replace `"http://your-dataiku-instance.com"` and `"your_api_key"` with your actual
Dataiku host and API key.
```

File: docs/knowledge_base/tips/dataiku/index.md [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+
+host = "http://your-dataiku-instance.com"
+apiKey = "your_api_key"
+
+client = DSSClient(host, apiKey)
+```
+
+## Retrieving Data from Dataiku
+
+Fetch data from your Dataiku projects by iterating through the projects and collecting relevant metrics for each model. This example code retrieves metrics like AUC and organizes them into a `pandas.DataFrame`.
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
Fetch data from your Dataiku projects by iterating through the projects and collecting
relevant metrics for each model. This example code retrieves metrics like AUC and organizes
them into a `pandas.DataFrame`.
```

File: docs/knowledge_base/tips/dataiku/index.md [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+        except Exception as e:
+            print(e)
+            pass
+
+metrics_df = pd.DataFrame(metrics_table)
+```
+
+## Preparing Data for Visualization
+
+Before visualization, you may need to further prepare your data. For instance, you can concatenate certain columns to form a unique identifier or name for each entry. Here, we add a 'Name' column to our DataFrame.
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
Before visualization, you may need to prepare your data further. For instance, you can
concatenate certain columns to form a unique identifier or name for each entry. Here, we add
a 'Name' column to our DataFrame.
```

File: docs/knowledge_base/tips/dataiku/index.md [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+
+```python
+metrics_df['Name'] = metrics_df[['Project', 'ModelId', 'MetricType']].agg(' - '.join, axis=1)
+```
+
+## Creating and Running a Taipy Application
+
+Define the GUI using Taipy's markdown-like syntax. This GUI will include dynamic selectors for projects and metrics, a table to display data, and a bar chart for visualizing metric values.
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
Define the GUI using Taipy's markdown-like syntax. This GUI will include dynamic selectors for projects and metrics, a table to display data, and a bar chart for visualizing metric values.
```

File: docs/knowledge_base/tips/dataiku/index.md [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+Gui(md).run()
+```
+
+
+
+# Dataiku Projects with Taipy Scenarios
+
+## Scenarios and Dataiku Integration
+
+Creating and executing projects on Dataiku involves several steps, from setting up your Dataiku instance to defining and running projects. Here's a step-by-step guide on how to seamlessly integrate Dataiku projects with Taipy scenarios:
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
Creating and executing projects on Dataiku involves several steps, from setting up your
Dataiku instance to defining and running projects. Here's a step-by-step guide on how to
seamlessly integrate Dataiku projects with Taipy scenarios:
```

File: docs/knowledge_base/tips/dataiku/index.md [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+
+- **Define Project Details:** Enter a name for your project and choose the type of project you wish to create (e.g., Visual Analysis, Data Preparation).
+
+**2 - Define Dataiku Project Logic**
+
+- **Prepare the Data:** Create datasets, recipes, and models required for your project in the Flow section.
+
+- **Write Code:** Write code for data processing, analysis, or any other tasks within notebooks or custom recipes.
+
+Here's an example of how parameters can be used in a Dataiku project to control the workflow and obtain results:
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]: In the article, I would assume these steps (1 and 2) as a prerequisite. It is already done in Dataiku.
We don't want to document how to create a Dataiku project. This is not our goal plus it's going to be painful to maintain.

File: docs/knowledge_base/tips/dataiku/index.md [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+output_dataset.write_with_schema(processed_df)
+```
+
+- **Test in Project:** Ensure that your workflow runs successfully within the project.
+
+**3 - Integrate with Taipy**
+
+- **API Calls or Plugins:** Use Dataiku's API or develop plugins to seamlessly integrate with Taipy scenarios for job triggering and result retrieval.
+
+By adapting the given Dataiku API script to work with a generic data node in Taipy, you can define reading and writing functions to interact with Dataiku datasets. Then, configure a generic data node to use these functions for data handling.
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
By adapting the given Dataiku API script to work with a generic data node in Taipy, you can
define reading and writing functions to interact with Dataiku datasets. Then, configure a
generic data node to use these functions for data handling.
```

File: docs/knowledge_base/tips/dataiku/index.md [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+
+<|{show_projects(metrics_df, selected_projects, selected_metrics)}|chart|x=Name|y=Value|type=bar|>
+"""
+
+Gui(md).run()
+```
+
+
+
+# Dataiku Projects with Taipy Scenarios
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]: Can you have a title showing the motivation ?

"Run a Dataiku pipeline from your Taipy application"
"Let your end-user run a Dataiku pipeline"
etc.

File: docs/knowledge_base/tips/dataiku/index.md [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md:138]: I would add a small introduction explaining the rough idea, which is encapsulating a Dataiku call to run a pipeline into a taipy Task.

File: docs/knowledge_base/tips/dataiku/index.md [hunk 13]
```diff
@@ -0,0 +1,255 @@
...
+    read_fct=custom_read_from_dataiku,
+    read_fct_args=[OUTPUT_DATASET_NAME, PROJECT_KEY, HOST, API_KEY],
+    write_fct=custom_write_to_dataiku,
+    write_fct_args=[OUTPUT_DATASET_NAME, PROJECT_KEY, HOST, API_KEY]
+)
+```
+
+### Simplifying the Task: Triggering Dataiku Flows
+
+With data nodes handling data autonomously, tasks now focus on action triggers, such as starting Dataiku flows. Let's assume we have a function `trigger_dataiku_flow` that initiates a specific flow within a Dataiku project:
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
With data nodes handling data autonomously, tasks now focus on action triggers, such as
starting Dataiku flows. Let's assume we have a function `trigger_dataiku_flow` that initiates
a specific flow within a Dataiku project:
```

File: docs/knowledge_base/tips/dataiku/index.md [hunk 14]
```diff
@@ -0,0 +1,255 @@
...
+
+```python
+# Scenario configuration incorporating the Dataiku flow trigger task
+dataiku_flow_scenario_cfg = Config.configure_scenario(
+    id="dataiku_flow_scenario",
+    task_configs=[trigger_flow_task_cfg]
+)
+```
+
+This scenario outlines a workflow where Taipy manages the overall process, including the initiation of Dataiku flows, with data nodes autonomously handling data exchanges with Dataiku.
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
This scenario outlines a workflow where Taipy manages the overall process, including the
initiation of Dataiku flows, with data nodes autonomously handling data exchanges with
Dataiku.
```

File: docs/knowledge_base/tips/dataiku/index.md [hunk 15]
```diff
@@ -0,0 +1,255 @@
...
+
+### Execution: Bringing It All Together
+
+To execute the scenario, use Taipy's execution model. This initiates the configured workflow, including autonomous data exchange with Dataiku and triggering the specified flow:
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
To execute the scenario, use Taipy's execution model. This initiates the configured workflow,
including autonomous data exchange with Dataiku and triggering the specified flow:
```

File: docs/knowledge_base/tips/dataiku/index.md [hunk 16]
```diff
@@ -0,0 +1,255 @@
...
+
+View all the information on output dataset here:
+
+<|{scenario.output_dataiku_dataset if scenario else None}|data_node|>
+"""
+
+    tp.Gui(scenario_md).run()
+```
+
+This streamlined approach, where tasks focus on action triggers (like starting Dataiku flows) while data nodes manage data reading and writing, enhances the interaction between the end user and its data and models.
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
This streamlined approach, where tasks focus on action triggers (like starting Dataiku flows)
while data nodes manage data reading and writing, enhancing the interaction between the 
end-user and its data and models.
```

File: docs/knowledge_base/tips/dataiku/index.md [hunk 17]
```diff
@@ -0,0 +1,255 @@
...
+
+In conclusion, integrating Dataiku with Taipy provides a powerful solution for streamlining data processing, visualization, and workflow orchestration. By leveraging Taipy's scenario management and Dataiku's robust capabilities, organizations can enhance efficiency, scalability, and collaboration in their data workflows. This integration enables seamless interaction between Dataiku projects and Taipy, facilitating the creation and scalability data-driven applications.
```
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
In conclusion, integrating Dataiku with Taipy provides a powerful solution for streamlining
data processing, visualization, and workflow orchestration. By leveraging Taipy's scenario
management and Dataiku's robust capabilities, organizations can enhance efficiency, scalability,
and collaboration in their data workflows. This integration enables seamless interaction between
Dataiku projects and Taipy, facilitating the creation and scalability of data-driven
applications.
```

**Comment by FlorianJacta** [comment]
New update of article:
- New code
- New structure
- Suggestions accepted
- Code available
- Screenshot

[Integrating with Dataiku - Taipy (2).zip](https://github.com/Avaiga/taipy-doc/files/14311040/Integrating.with.Dataiku.-.Taipy.2.zip)

image

image

**Comment by FlorianJacta** [comment]
Waiting @marisogo for illustration

Done!

**Review by jrobinAV — APPROVED** [review]

**Comment by FlorianJacta** [comment]
Waiting for Dataiku approval

**Comment by jrobinAV** [comment]
> Waiting for Dataiku approval

I am adding the label blocked until we can proceed.

**Comment by jrobinAV** [comment]
@FlorianJacta, Any news on that?

**Comment by FlorianJacta** [comment]
Waiting Dataiku answer still

**Comment by jrobinAV** [comment]
@FlorianJacta Can we delete the branch?

**Pull request merged**